### PR TITLE
DOC-4202 exact match search examples

### PR DESCRIPTION
**Pull request by andy-stark-redis**
Java versions of the examples in [this page](https://redis.io/docs/latest/develop/interact/search-and-query/query/exact-match/).

### CODE REVIEW

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+        try {
+            SearchResult res4 = jedis.ftSearch(
+                "idx:email",
+                "test@redis.com",
+                new FTSearchParams().dialect(2)
+            );
+            System.out.println(res4.getTotalResults());
+        } catch (JedisDataException j) {
+            System.out.println("Query syntax 'test@redis.com' not supported in Jedis.");
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:311]: Did you try helper methods in RediSearchUtil?

Comment (andy-stark-redis) [reply src/test/java/io/redis/examples/QueryEmExample.java]: @sazzad16 I didn't know about RediSearchUtil until now, but I've tried it and it works fine. Thanks for the advice!

Comment (sazzad16) [reply src/test/java/io/redis/examples/QueryEmExample.java]: @andy-stark-redis No problem! Happy to help :slightly_smiling_face:

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+import redis.clients.jedis.UnifiedJedis;
+import redis.clients.jedis.json.Path2;
+import redis.clients.jedis.search.FTCreateParams;
+import redis.clients.jedis.search.FTSearchParams;
+import redis.clients.jedis.search.IndexDataType;
+import redis.clients.jedis.search.Query;
+import redis.clients.jedis.search.schemafields.*;
+import redis.clients.jedis.search.SearchResult;
+import redis.clients.jedis.search.Document;
+import redis.clients.jedis.search.RediSearchUtil;
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:20]:
```suggestion
import redis.clients.jedis.search.*
import redis.clients.jedis.search.schemafields.*;
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+        // Tests for 'em4' step.
+        // REMOVE_START
+        Assert.assertEquals(1, res5.getTotalResults());
+        Assert.assertEquals("bicycle:8", docs5.get(0).getId());
+        // REMOVE_END
+
+
+// HIDE_START
+    }
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:331]:
```suggestion
        // REMOVE_END

// HIDE_START
        jedis.close();
    }
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+            TextField.of("$.model").as("model"),
+            TextField.of("$.description").as("description"),
+            NumericField.of("$.price").as("price"),
+            TagField.of("$.condition").as("condition")
+        };
+
+        jedis.ftCreate("idx:bicycle",
+            FTCreateParams.createParams()
+                .on(IndexDataType.JSON)
+                .addPrefix("bicycle:"),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:45]:
```suggestion
            FTCreateParams.createParams()
                    .on(IndexDataType.JSON)
                    .addPrefix("bicycle:"),
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+            + "suggest taking it to the mountains. Fenders protect you from mud, and a rear "
+            + "basket lets you transport groceries, flowers and books. The ThrillCycle comes "
+            + "with a limited lifetime warranty, so this little guy will last you long "
+            + "past graduation.\","
+            + "    \"condition\": \"refurbished\""
+            + "  }"
+        };
+
+        for (int i = 0; i < bicycleJsons.length; i++) {
+            jedis.jsonSet("bicycle:" + i, new Path2("$"), bicycleJsons[i]);
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:211]:
```suggestion
            jedis.jsonSet("bicycle:" + i, Path2.ROOT_PATH, bicycleJsons[i]);
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+        }
+// HIDE_END
+
+
+        // STEP_START em1
+        SearchResult res1 = jedis.ftSearch("idx:bicycle","@price:[270 270]");
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:217]:
```suggestion
        SearchResult res1 = jedis.ftSearch("idx:bicycle", "@price:[270 270]");
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+        List<Document> docs1 = res1.getDocuments();
+
+        for (int i = 0; i < docs1.size(); i++) {
+            System.out.println(docs1.get(i).getId());
+        }
+        // >>> bicycle:0
+
+        SearchResult res2 = jedis.ftSearch(
+            "idx:bicycle",
+            "*",
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:229]:
```suggestion
        SearchResult res2 = jedis.ftSearch("idx:bicycle",
            "*",
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+            FTSearchParams.searchParams()
+                .filter("price", 270, 270)
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:231]:
```suggestion
            FTSearchParams.searchParams()
                    .filter("price", 270, 270)
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+
+        // STEP_START em3
+        SchemaField[] emailSchema = {
+            TextField.of("$.email").as("email")
+        };
+
+        jedis.ftCreate("idx:email", 
+            new FTCreateParams()
+                .addPrefix("key:")
+                .on(IndexDataType.JSON),
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:288]:
```suggestion
            new FTCreateParams()
                    .addPrefix("key:")
                    .on(IndexDataType.JSON),
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+            emailSchema
+        );
+
+        jedis.jsonSet("key:1", new Path2("$"), "{\"email\": \"test@redis.com\"}");
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:292]:
```suggestion
        jedis.jsonSet("key:1", Path2.ROOT_PATH, "{\"email\": \"test@redis.com\"}");
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+        
+        SearchResult res4 = jedis.ftSearch(
+            "idx:email",
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:295]:
```suggestion
        SearchResult res4 = jedis.ftSearch("idx:email",
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 12]
```diff
@@ -0,0 +1,334 @@
...
+
+        // Tests for 'em3' step.
+        // REMOVE_START
+        jedis.ftDropIndex("idx:email");
+        // REMOVE_END
+
+
+        // STEP_START em4
+        SearchResult res5 = jedis.ftSearch(
+            "idx:bicycle",
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:310]:
```suggestion
        SearchResult res5 = jedis.ftSearch("idx:bicycle",
```

File: src/test/java/io/redis/examples/QueryEmExample.java [hunk 13]
```diff
@@ -0,0 +1,334 @@
...
+            new Query("@description:\"rough terrain\"")
```
Comment (sazzad16) [inline src/test/java/io/redis/examples/QueryEmExample.java:311]: Using Query...
Should we keep all the same?
Using a different style (Query) also makes sense as it informs users about it.

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**